code/modules/reagents/chemistry/reagents_holder.dm

```diff
@@ -248,6 +248,20 @@
 
 	handle_reactions()
 
+// This proc increases the temp of reagents by `temp_increase` amount, up until the target temperature, and no more
+/datum/reagents/proc/change_reagent_temp(temp_increase, target_temp)
+	var/difference
+	if(abs(target_temp - chem_temp) > temp_increase)
+		difference = temp_increase
+	else
+		difference = target_temp - chem_temp
+	if(target_temp > chem_temp)
+		chem_temp += difference
+	else if(target_temp < chem_temp)
+		chem_temp -= difference
```

Comment:
You'll probably want to clamp the adjusted `chem_temp` between `temperature_max`/`_min` like `temperature_reagents` does (but use `clamp(...)` instead of `max(min(...))`